tracker_test.go

```diff
@@ -9,65 +9,85 @@ import (
 	"github.com/almighty/almighty-core/resource"
 )
 
+type trackerAttr struct {
+	Type string
+	URL  string
+}
+
+func getTrackerPayload(attr trackerAttr) app.CreateTrackerPayload {
+	return app.CreateTrackerPayload{
+		Data: &app.TrackerData{
+			Type: APIStringTypeTracker,
+			Attributes: &app.TrackerAttributes{
+				Type: attr.Type,
+				URL:  attr.URL,
+			},
+		},
+	}
+}
+
 func TestCreateTracker(t *testing.T) {
 	resource.Require(t, resource.Database)
 	controller := TrackerController{Controller: nil, db: gormapplication.NewGormDB(DB), scheduler: RwiScheduler}
-	payload := app.CreateTrackerAlternatePayload{
+
+	payload := getTrackerPayload(trackerAttr{
 		URL:  "http://issues.jboss.com",
 		Type: "jira",
-	}
-
+	})
 	_, created := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
-	if created.ID == "" {
+	if *created.Data.ID == "" {
 		t.Error("no id")
 	}
 }
 
 func TestGetTracker(t *testing.T) {
 	resource.Require(t, resource.Database)
 	controller := TrackerController{Controller: nil, db: gormapplication.NewGormDB(DB), scheduler: RwiScheduler}
-	payload := app.CreateTrackerAlternatePayload{
+	payload := getTrackerPayload(trackerAttr{
 		URL:  "http://issues.jboss.com",
 		Type: "jira",
-	}
+	})
 
 	_, result := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
-	test.ShowTrackerOK(t, nil, nil, &controller, result.ID)
-	_, tr := test.ShowTrackerOK(t, nil, nil, &controller, result.ID)
+	resultID := *result.Data.ID
+	test.ShowTrackerOK(t, nil, nil, &controller, resultID)
+	_, tr := test.ShowTrackerOK(t, nil, nil, &controller, resultID)
 	if tr == nil {
-		t.Fatalf("Tracker '%s' not present", result.ID)
+		t.Fatalf("Tracker '%s' not present", resultID)
 	}
-	if tr.ID != result.ID {
-		t.Errorf("Id should be %s, but is %s", result.ID, tr.ID)
+	if tr.ID != resultID {
+		t.Errorf("Id should be %s, but is %s", resultID, tr.ID)
 	}
 
 	payload2 := app.UpdateTrackerAlternatePayload{
 		URL:  tr.URL,
 		Type: tr.Type,
 	}
 	_, updated := test.UpdateTrackerOK(t, nil, nil, &controller, tr.ID, &payload2)
-	if updated.ID != result.ID {
-		t.Errorf("Id has changed from %s to %s", result.ID, updated.ID)
+	if updated.ID != resultID {
+		t.Errorf("Id has changed from %s to %s", resultID, updated.ID)
 	}
-	if updated.URL != result.URL {
-		t.Errorf("URL has changed from %s to %s", result.URL, updated.URL)
+	resultURL := result.Data.Attributes.URL
+	if updated.URL != resultURL {
+		t.Errorf("URL has changed from %s to %s", resultURL, updated.URL)
 	}
-	if updated.Type != result.Type {
-		t.Errorf("Type has changed has from %s to %s", result.Type, updated.Type)
+	resultType := result.Data.Attributes.Type
+	if updated.Type != resultType {
+		t.Errorf("Type has changed has from %s to %s", resultType, updated.Type)
 	}
 
-	test.DeleteTrackerOK(t, nil, nil, &controller, result.ID)
+	test.DeleteTrackerOK(t, nil, nil, &controller, resultID)
```

Comment:
Use `gormsupport.DeleteCreatedEntities` as a deferred call. There are many examples in other files.